lib/auxiliary/sysinfo_parser.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'active_support'
+require 'active_support/core_ext/object/deep_dup'
+
+# AutoHCK module
+module AutoHCK
+  # UtilsParser class
+  class UtilsParser
+    # Converts string and float input to int. Strips all non-numeric
+    # characters from strings.
+    #
+    # @param value [String, Float] Input value
+    #
+    # @return [Integer, nil] Integer if successful conversion, otherwise nil
+    def self.convert_to_int(value)
+      if value.is_a?(String)
+        str_val = value.gsub(/[^0-9\-.]/, '')
+        begin
+          Integer(str_val)
+        rescue ArgumentError, TypeError
+          begin
+            Integer(Float(str_val))
+          rescue ArgumentError, TypeError
+            nil
+          end
+        end
+      elsif value.is_a?(Numeric)
+        value.to_i
+      end
+    end
+  end
+
+  # SysInfoParser class
+  class SysInfoParser
+    DELIM = ':' # k/v delimiter
+
+    # Main text parsing function
+    #
+    # @param data [String] Text data to parse
+    #
+    # @return [Array<Hash>] List of dictionaries. Processed structured data.
+    def parse(data)
+      rows = CSV.parse(data, headers: true)
+      raw_data = rows.first.to_h
+
+      raw_output = {}
+      raw_data.each do |k, v|
+        raw_output[k] = case k
+                        when 'Hotfix(s)', 'Processor(s)'
+                          parse_hotfixs_or_processors(v)
+                        else
+                          v.strip
```

Comment:
Is this `strip` really necessary?

Reply:
Yes. From one PC `"Microsoft Windows Server 2008 R2 Standard "`